prefect_aws/client_waiter.py

```diff
@@ -0,0 +1,100 @@
+"""Task for waiting on a long-running AWS job"""
+
+import json
+from typing import Any, Dict, Optional
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    import importlib_resources as pkg_resources
+
+from botocore.waiter import WaiterModel, create_waiter_with_client
+from prefect import get_run_logger, task
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+
+from prefect_aws import waiters
+from prefect_aws.credentials import AwsCredentials
+
+
+@task
+async def client_waiter(
+    client: str,
+    waiter_name: str,
+    aws_credentials: AwsCredentials,
+    waiter_definition: Optional[Dict[str, Any]] = None,
+    **waiter_kwargs: Optional[Dict[str, Any]],
+):
+    """
+    Uses the underlying boto3 waiter functionality.
+
+    Args:
+        client: The AWS client on which to wait (e.g., 'client_wait', 'ec2', etc).
+        waiter_name: The name of the waiter to instantiate.
+            Can be a boto-supported
+            waiter or one of prefect's custom waiters.
+            Currently, Prefect offers three additional
+            waiters for AWS client_wait:
+
+            - `"JobExists"` waits for a job to be instantiated
+            - `"JobRunning"` waits for a job to start running
+            - `"JobComplete"` waits for a job to finish.
+
+            You can find the definitions for all prefect-defined waiters
+            [here](https://github.com/PrefectHQ/prefect-aws/
```

Comment:
I think we should document the available Prefect waiters in in `prefect_aws/waiters/__init.py` so that they show up in the docs.